pandas/tests/indexes/test_base.py

```diff
@@ -707,6 +707,98 @@ def test_intersect_str_dates(self):
 
         assert len(res) == 0
 
+    def create_empty_index(self, id):
```

Comment:
what is this for?

Reply:
I need a way to have an empty index that is independent of the individual `Index` class that is tested in `test_corner_union`.  Alternatively, I could just call `id.drop(id)` each time I need to create an empty index for the test.  IMHO (and I can be convinced otherwise), having the calls to `_create_empty_index` in the test makes it clear that an empty index is being used in the test.